webapp/tests/karma/ts/services/rules-engine.service.spec.ts

```diff
@@ -72,6 +74,9 @@ describe('RulesEngineService', () => {
     type: 'user-settings',
     roles: [],
   };
+  const chtScriptApi = {
+    v1: { hasPermission: () => {} }
```

Comment:
Please use `sinon.stub()` instead of noop functions.

Reply:
Fixed!